modules/phase_field/include/utils/ExpressionBuilder.h

```diff
@@ -399,6 +401,24 @@ class ExpressionBuilder
     BINARY_OP_IMPLEMENT(==,EQ)
     BINARY_OP_IMPLEMENT(!=,NOTEQ)
 
+    /*
+     * Compound assignment operators
+     */
+    #define BINARYCOMP_OP_IMPLEMENT(op,OP) \
+    EBTerm & operator op (const EBTerm & term) { \
+      mooseAssert(_root != NULL, "Empty term provided on left side of operator " #op); \
+      mooseAssert(term._root != NULL, "Empty term provided on right side of operator " #op); \
+      if (dynamic_cast<EBTempIDNode *>(_root)) \
```

Comment:
Terms that are uninitialized are assigned a unique identifier that allows them to be used in another term without holding any value for being substituted out later on. Using a compound assignment on such a temporary node is not supported yet. This might be useful in the future if/when the substitution system is made more powerful to support substitution of arbitrary terms (rather than just symbols and temporary id terms).